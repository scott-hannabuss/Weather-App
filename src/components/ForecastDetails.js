import React from "react";
import PropTypes from "prop-types";
import moment from "moment";

const ForecastDetails = ({ forecast }) => {
  const { date, temperature, wind, humidity } = forecast;
  return (
    <div className="forecast-details" data-testid="forecast-details">
      <div className="forecast-details__date">
        {moment(date).format("ddd Do MMM")}
      </div>
      <div className="forecast-details__min-temperature">
        Min temp:
        {temperature.min}
        &deg;C
      </div>
      <div className="forecast-details__max-temperature">
        Max temp:
        {temperature.max}
        &deg;C
      </div>
      <div className="forecast-details__humidity">
        Humidity:
        {humidity}
        &#17;
      </div>
      <div className="forecast-details__wind-speed">
        Wind speed:
        {wind.speed}
        mph
      </div>
      <div className="forecast-details__wind-direction">
        Wind direction:
        {wind.direction}
      </div>
    </div>
  );
};

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecast: PropTypes.shape({
    date: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temperature: PropTypes.shape({
      max: PropTypes.number,
      min: PropTypes.number,
    }).isRequired,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }).isRequired,
  }).isRequired,
};
