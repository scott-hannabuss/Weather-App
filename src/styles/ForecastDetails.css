@import url("https://fonts.googleapis.com/css2?family=Open+Sans+Condensed:wght@300&display=swap");

.forecast-details {
  color: white;
  justify-content: center;
  align-content: center;
  margin-right: 20%;
  margin-left: 20%;
  display: grid;
  padding-top: 2%;
  padding-bottom: 2%;
  border: 1px solid;
  transform: scaleX(1.3);
  font-family: "Open Sans Condensed", sans-serif;
}

@media only screen and (min-width: 751px) and (max-width: 1025px) {
  .forecast-details {
    margin-top: 7.5%;
  }
}
@media only screen and (max-width: 750px) {
  .forecast-details {
    margin-top: 10%;
  }
}
