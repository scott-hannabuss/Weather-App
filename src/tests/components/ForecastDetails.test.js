import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 11111111,
    humidity: 80,
    temperature: {
      min: 5,
      max: 30,
    },
    wind: {
      speed: 30,
      direction: "sw",
    },
  };

  it("renders as expected", () => {
    const { asFragment } = render(<ForecastDetails forecast={validProps} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
