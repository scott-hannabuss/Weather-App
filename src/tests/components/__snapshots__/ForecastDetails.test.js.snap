// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders as expected 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Thu 1st Jan
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      Min temp:5°C
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      Max temp:30°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity:80
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind speed:30mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind direction:sw
    </div>
  </div>
</DocumentFragment>
`;
